echobot/echobot.c

```diff
@@ -18,22 +18,111 @@ typedef struct DHT_node {
     const char *ip;
     uint16_t port;
     const char key_hex[TOX_PUBLIC_KEY_SIZE * 2 + 1];
-    unsigned char key_bin[TOX_PUBLIC_KEY_SIZE];
 } DHT_node;
 
+const char *savedata_filename = "savedata.tox";
```

Comment:
`static const char *const savedata_filename`

Reply:
What about functions being non-static or `Tox *create_tox()` not being `Tox *create_tox(void)`,
or not checking return values of almost all functions that can fail, or using  [non portable](https://wiki.sei.cmu.edu/confluence/display/c/FIO19-C.+Do+not+use+fseek%28%29+and+ftell%28%29+to+compute+the+size+of+a+regular+file) `fseek(f, 0, SEEK_END);`, assigning an int to a char `tox_id_hex[i] = toupper(tox_id_hex[i]);` , or...

well, there are many things wrong with this code, however that's not the point. The point is for it to be readable, with as little unnecessary error checking, typing and casting clutter as possible, to serve as a tutorial on how to get started with tox and write a simple tox client.

Reply:
Well, at least that's the point of it on the wiki article, and the echobot was taken here from the wiki, so i'd like to keep them in sync.